src/main/java/org/apereo/cas/config/CasOAuth20ServicesConfiguration.java

```diff
@@ -43,8 +45,11 @@
 
     @Bean
     public Service oauthCallbackService() {
-        val oAuthCallbackUrl = casProperties.getServer().getPrefix()
-                + OAuth20Constants.BASE_OAUTH20_URL + '/' + OAuth20Constants.CALLBACK_AUTHORIZE_URL_DEFINITION;
+        // OSF CAS customization: there is no functionality change to this configuration class (as of 6.2.8). However, it must overlaid
+        // so that the Spring configuration process can use the overlaid interface constant `OAuth20Constants.BASE_OAUTH20_URL` correctly.
+        // This is similar to the issue in `CasOAuth20Configuration`, where interface constants must either 1) be directly used instead of
+        // via a built-in helper method or 2) be used via a customized helper method.
+        val oAuthCallbackUrl = OsfCasOAuth20Utils.casOAuthCallbackUrlDefinition(casProperties.getServer().getPrefix());
```

Comment:
The vanilla way of building the call back URL definition using string concatenation **WORKS** as long as this configuration class `CasOAuth20ServicesConfiguration` is overlaid. However, I decided to replace it with and customized helper methods provided by `OsfCasOAuth20Utils`, which is similar to (and consistent with)what the other configuration class `CasOAuth20Configuration.java` does.